app/api/entities/specs/entitySavingManager.spec.ts

```diff
@@ -329,13 +373,197 @@ describe('entitySavingManager', () => {
         });
 
         expect(savedEntity._id).not.toBeNull();
         expect(savedEntity.metadata.text[0].value).toBe('a text');
         expect(savedEntity.metadata.image).toEqual([]);
       });
     });
+
+    describe('entity with main documents', () => {
+      let savedEntity: EntityWithFilesSchema;
+      let expectedDocuments: FileType[];
+
+      beforeEach(() => {
+        savedEntity = {};
+        expectedDocuments = [];
+      });
+
+      const expectCheck = (entity: EntityWithFilesSchema) => {
+        expect(entity.documents).toMatchObject(expectedDocuments);
+      };
+
+      const emiter = jest
+        .fn()
+        .mockImplementation(event => event === 'documentProcessed' && expectCheck(savedEntity));
+
+      it('should create an entity with main documents', async () => {
+        const entity = {
+          title: 'newEntity',
+          template: template1Id,
+        };
+
+        expectedDocuments = [
+          {
+            mimetype: 'application/pdf',
+            originalname: 'myNewFile.pdf',
+            size: 12,
+            type: 'document',
+          },
+        ];
+
+        ({ entity: savedEntity } = await saveEntity(entity, {
+          ...reqData,
+          socketEmiter: emiter,
+          files: [{ ...newMainPdfDocument, fieldname: 'documents[0]' }],
+        }));
+      });
+
+      describe('updating an entity', () => {
+        it('should keep existing documents', async () => {
+          const entity = {
+            _id: entity3Id,
+            sharedId: 'shared3',
+            title: 'entity3',
+            template: template1Id,
+          };
+
+          expectedDocuments = [
+            {
+              entity: 'shared3',
+              mimetype: 'application/pdf',
+              originalname: 'Sample main PDF File.pdf',
+              type: 'document',
+            },
+            {
+              entity: 'shared3',
+              mimetype: 'application/pdf',
+              originalname: 'myNewFile.pdf',
+              type: 'document',
+            },
+          ];
+
+          ({ entity: savedEntity } = await saveEntity(entity, {
+            ...reqData,
+            files: [{ ...newMainPdfDocument, fieldname: 'documents[0]' }],
+            socketEmiter: emiter,
+          }));
+
+          expect(savedEntity.attachments).toMatchObject([
+            {
+              mimetype: 'text/plain',
+              originalname: 'Sample Text File.txt',
+              filename: 'samplefile.txt',
+              type: 'attachment',
+            },
+          ]);
+        });
+
+        it('should remove files and thumbnails for deleted documents', async () => {
+          const entity = {
+            _id: entity3Id,
+            sharedId: 'shared3',
+            title: 'entity3',
+            template: template1Id,
+            attachments: [entity3textFile],
+          };
+
+          expectedDocuments = [];
+
+          ({ entity: savedEntity } = await saveEntity(entity, {
```

Comment:
Should this have the emitter as well, or is that redundant? Same question for the next text.

Reply:
We are using an empty function as the default value for the emitter, but yes, to be consistent in all the calls it should be the mocked function. 👍